### PR TITLE
Realized own the simple web application.

### PR DESCRIPTION
**Pull request by gigbat**

### CODE REVIEW

File: src/main/java/internet/shop/controllers/cars/AddCarController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        req.getRequestDispatcher("/WEB-INF/views/cars/create/addCar/create_car.jsp")
+                .forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Manufacturer manufacturer = manufacturerService
+                .get(Long.parseLong(req.getParameter("manufacturer_id")));
+        String model = req.getParameter("model");
```
Comment (Ketch92) [inline src/main/java/internet/shop/controllers/cars/AddCarController.java:33]: what if passed id isn't in the DB?

File: src/main/java/internet/shop/controllers/cars/AddCarController.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+        Car car = new Car();
+        car.setManufacturer(manufacturer);
+        car.setModel(model);
+
+        carService.create(car);
+        resp.sendRedirect("/");
```
Comment (Ketch92) [inline src/main/java/internet/shop/controllers/cars/AddCarController.java]: a bit of hardcoding

File: src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/cars/create"
+                + "/addDriverToCar/addDriverToCar.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        Long driverId = Long.valueOf(request.getParameter("driver_id"));
+        Long carId = Long.valueOf(request.getParameter("car_id"));
```
Comment (Ketch92) [inline src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java:32]: as well what if ids aren't in the DB

File: src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+        Car car = carService.get(carId);
+        Driver driver = driverService.get(driverId);
+
+        carService.addDriverToCar(driver, car);
+        response.sendRedirect("/");
```
Comment (Ketch92) [inline src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java]: same hardcoding

**Review by Ketch92 — CHANGES_REQUESTED** [review]

File: src/main/java/internet/shop/controllers/cars/AddCarController.java (hunk 1 above)
Comment (gigbat) [reply src/main/java/internet/shop/controllers/cars/AddCarController.java:33]: done

File: src/main/java/internet/shop/controllers/cars/AddCarController.java (hunk 2 above)
Comment (gigbat) [reply src/main/java/internet/shop/controllers/cars/AddCarController.java]: done

File: src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java (hunk 3 above)
Comment (gigbat) [reply src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java:32]: done

File: src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java (hunk 4 above)
Comment (gigbat) [reply src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java]: done

**Review by Ketch92 — APPROVED** [review]

File: src/main/java/internet/shop/controllers/cars/AddCarController.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    private static final Injector injector = Injector.getInstance("internet.shop");
+    private CarService carService = (CarService) injector
+            .getInstance(CarService.class);
+    private ManufacturerService manufacturerService = (ManufacturerService) injector
+            .getInstance(ManufacturerService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/car/create/car/create_car.jsp")
```
Comment (KseniiaMakarova) [inline src/main/java/internet/shop/controllers/cars/AddCarController.java]:
```suggestion
        req.getRequestDispatcher("/WEB-INF/views/car/create.jsp")
```
do we need such complex folder tree here?

File: src/main/java/internet/shop/controllers/cars/AddCarController.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Manufacturer manufacturer = null;
+        try {
+            manufacturer = manufacturerService
+                    .get(Long.parseLong(req.getParameter("manufacturer_id")));
+        } catch (RuntimeException e) {
```
Comment (KseniiaMakarova) [inline src/main/java/internet/shop/controllers/cars/AddCarController.java]: it is not a good practice to catch RuntimeExceptions
let's remove try-catch here and assume that we will receive only valid ids here

File: src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+            throws ServletException, IOException {
+        Long driverId = Long.valueOf(request.getParameter("driver_id"));
+        Long carId = Long.valueOf(request.getParameter("car_id"));
+
+        Car car = null;
+        Driver driver = null;
+        try {
+            car = carService.get(carId);
+            driver = driverService.get(driverId);
+        } catch (RuntimeException e) {
```
Comment (KseniiaMakarova) [inline src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java]: same here

File: src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    private final CarService carService = (CarService) injector
+            .getInstance(CarService.class);
+    private final DriverService driverService = (DriverService) injector
+            .getInstance(DriverService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/car/create"
+                + "/driver/addDriverToCar.jsp").forward(req, resp);
```
Comment (KseniiaMakarova) [inline src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java]:
```suggestion
        req.getRequestDispatcher("/WEB-INF/views/car/addDriverToCar.jsp").forward(req, resp);
```

File: src/main/java/internet/shop/controllers/drivers/AddDriverController.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+public class AddDriverController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("internet.shop");
+    private DriverService driverService = (DriverService) injector
+            .getInstance(DriverService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/driver/create/create_driver.jsp")
```
Comment (KseniiaMakarova) [inline src/main/java/internet/shop/controllers/drivers/AddDriverController.java]:
```suggestion
        req.getRequestDispatcher("/WEB-INF/views/driver/create.jsp")
```

File: src/main/java/internet/shop/controllers/manufacturers/AddManufacturerController.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+public class AddManufacturerController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("internet.shop");
+    private ManufacturerService manufacturerService = (ManufacturerService) injector
+            .getInstance(ManufacturerService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/manufacturer/create"
+                + "/create_manufacturer.jsp").forward(req, resp);
```
Comment (KseniiaMakarova) [inline src/main/java/internet/shop/controllers/manufacturers/AddManufacturerController.java]:
```suggestion
        req.getRequestDispatcher("/WEB-INF/views/manufacturer/create.jsp").forward(req, resp);
```

File: src/main/webapp/WEB-INF/views/index.jsp [hunk 11]
```diff
@@ -1,9 +1,37 @@
...
+<a href="${pageContext.request.contextPath}/manufacturer/create">Page to create manufacturer</a>
+<a href="${pageContext.request.contextPath}/manufacturer">Page to get all manufacturer</a>
+<a href="${pageContext.request.contextPath}/driver/create">Page to create driver</a>
+<a href="${pageContext.request.contextPath}/driver">Page to get all driver</a>
+<a href="${pageContext.request.contextPath}/car/create/car">Page to create car</a>
+<a href="${pageContext.request.contextPath}/car/create/driver">Page to add driver to car</a>
+<a href="${pageContext.request.contextPath}/car">Page to get all car</a>
 </body>
-</html>
+</html>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/index.jsp]: add empty line to the end of the file

File: src/main/webapp/web.xml [hunk 12]
```diff
@@ -11,6 +11,106 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>createManufacturer</servlet-name>
+        <servlet-class>internet.shop.controllers.manufacturers.AddManufacturerController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>createManufacturer</servlet-name>
+        <url-pattern>/manufacturer/create</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/manufacturers/create</url-pattern>
```
it is good practice to use plurals in URLs

File: src/main/webapp/web.xml [hunk 13]
```diff
@@ -11,6 +11,106 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>createDriver</servlet-name>
+        <servlet-class>internet.shop.controllers.drivers.AddDriverController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>createDriver</servlet-name>
+        <url-pattern>/driver/create</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers/create</url-pattern>
```

File: src/main/webapp/web.xml [hunk 14]
```diff
@@ -11,6 +11,106 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>createCar</servlet-name>
+        <servlet-class>internet.shop.controllers.cars.AddCarController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>createCar</servlet-name>
+        <url-pattern>/car/create/car</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cars/create</url-pattern>
```

File: src/main/webapp/web.xml [hunk 15]
```diff
@@ -11,6 +11,106 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>getAllManufacturer</servlet-name>
+        <servlet-class>internet.shop.controllers.manufacturers.GetAllManufacturerController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>getAllManufacturer</servlet-name>
+        <url-pattern>/manufacturer</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/manufacturers</url-pattern>
```

File: src/main/webapp/web.xml [hunk 16]
```diff
@@ -11,6 +11,106 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>getAllDriver</servlet-name>
+        <servlet-class>internet.shop.controllers.drivers.GetAllDriverController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>getAllDriver</servlet-name>
+        <url-pattern>/driver</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers</url-pattern>
```

File: src/main/webapp/web.xml [hunk 17]
```diff
@@ -11,6 +11,106 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>getAllCar</servlet-name>
+        <servlet-class>internet.shop.controllers.cars.GetAllCarController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>getAllCar</servlet-name>
+        <url-pattern>/car</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cars</url-pattern>
```

File: src/main/webapp/web.xml [hunk 18]
```diff
@@ -11,6 +11,106 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>addDriverToCar</servlet-name>
+        <servlet-class>internet.shop.controllers.cars.AddDriverToCarController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>addDriverToCar</servlet-name>
+        <url-pattern>/car/create/driver</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cars/drivers/add</url-pattern>
```

File: src/main/webapp/web.xml [hunk 19]
```diff
@@ -11,6 +11,106 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteManufacturer</servlet-name>
+        <servlet-class>internet.shop.controllers.manufacturers.DeleteManufacturerController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>deleteManufacturer</servlet-name>
+        <url-pattern>/manufacturer/delete</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/manufacturers/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 20]
```diff
@@ -11,6 +11,106 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteDriver</servlet-name>
+        <servlet-class>internet.shop.controllers.drivers.DeleteDriverController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>deleteDriver</servlet-name>
+        <url-pattern>/driver/delete</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 21]
```diff
@@ -11,6 +11,106 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteCar</servlet-name>
+        <servlet-class>internet.shop.controllers.cars.DeleteCarController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>deleteCar</servlet-name>
+        <url-pattern>/car/delete</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cars/delete</url-pattern>
```

File: src/main/webapp/WEB-INF/views/car/get.jsp [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+<html>
+<head>
+    <title>Get all Cars</title>
+</head>
+<body>
+<table border="1">
+    <tr>
+        <th>ID</th>
+        <th>Name</th>
+        <th>License Number</th>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/car/get.jsp]: let's fix these column names for car

File: src/main/webapp/WEB-INF/views/car/get.jsp [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+        <th>Driver name</th>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/car/get.jsp]:
```suggestion
        <th>Driver names</th>
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/internet/shop/controllers/cars/AddCarController.java (hunk 5 above)
Comment (gigbat) [reply src/main/java/internet/shop/controllers/cars/AddCarController.java]: done, i think that i did the correct

File: src/main/java/internet/shop/controllers/cars/AddCarController.java (hunk 6 above)
Comment (gigbat) [reply src/main/java/internet/shop/controllers/cars/AddCarController.java]: done

File: src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java (hunk 7 above)
Comment (gigbat) [reply src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java]: done

File: src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java (hunk 8 above)
Comment (gigbat) [reply src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java]: done

File: src/main/java/internet/shop/controllers/drivers/AddDriverController.java (hunk 9 above)
Comment (gigbat) [reply src/main/java/internet/shop/controllers/drivers/AddDriverController.java]: done

File: src/main/java/internet/shop/controllers/manufacturers/AddManufacturerController.java (hunk 10 above)
Comment (gigbat) [reply src/main/java/internet/shop/controllers/manufacturers/AddManufacturerController.java]: done

File: src/main/webapp/WEB-INF/views/index.jsp (hunk 11 above)
Comment (gigbat) [reply src/main/webapp/WEB-INF/views/index.jsp]: done

File: src/main/webapp/web.xml (hunk 12 above)
Comment (gigbat) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 13 above)
Comment (gigbat) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 14 above)
Comment (gigbat) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 15 above)
Comment (gigbat) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 16 above)
Comment (gigbat) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 17 above)
Comment (gigbat) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 18 above)
Comment (gigbat) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 19 above)
Comment (gigbat) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 20 above)
Comment (gigbat) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 21 above)
Comment (gigbat) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/WEB-INF/views/car/get.jsp (hunk 22 above)
Comment (gigbat) [reply src/main/webapp/WEB-INF/views/car/get.jsp]: done

File: src/main/webapp/WEB-INF/views/car/get.jsp (hunk 23 above)
Comment (gigbat) [reply src/main/webapp/WEB-INF/views/car/get.jsp]: done

File: src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java [hunk 24]
```diff
@@ -0,0 +1,40 @@
...
+    private static final Injector injector = Injector.getInstance("internet.shop");
+    private final CarService carService = (CarService) injector
+            .getInstance(CarService.class);
+    private final DriverService driverService = (DriverService) injector
+            .getInstance(DriverService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/cars/addDriverToCar.jsp")
```
Comment (KseniiaMakarova) [inline src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java:24]:
```suggestion
        req.getRequestDispatcher("/WEB-INF/views/cars/add_driver_to_car.jsp")
```

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java (hunk 24 above)
Comment (gigbat) [reply src/main/java/internet/shop/controllers/cars/AddDriverToCarController.java:24]: done

**Pull request merged**